src/pages/CheckinPage.jsx

```diff
@@ -72,12 +81,15 @@ const CheckinPage = () => {
       const options = {
         keys: ['first_name', 'last_name', 'email'],
       };
-      const fuse = new Fuse(searchResults, options);
+      const fuse = new Fuse(
+        joinedData.filter(item => item.event_id == eventId || eventId == -1),
+        options,
+      );
       const searchResult = fuse.search(input);
       const reduceResult = searchResult.map(result => result.item);
       setVolunteerResults(reduceResult);
     }
-  }, [input, searchResults, joinedData]);
+  }, [input, joinedData]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'eventId'. Either include it or remove the dependency array.